core/iwasm/compilation/aot_emit_gc.c

```diff
@@ -104,4 +136,1476 @@ aot_call_aot_create_func_obj(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
+
+    GET_REF_FROM_STACK(gc_obj);
+
+    /* Check if gc object is NULL */
+    BUILD_ISNULL(gc_obj, cmp_gc_obj, "cmp_gc_obj");
+
+    ADD_BASIC_BLOCK(check_gc_obj_succ, "check_gc_obj_succ");
+    MOVE_BLOCK_AFTER_CURR(check_gc_obj_succ);
+
+    /*  Throw exception if it is NULL */
+    if (!aot_emit_exception(comp_ctx, func_ctx, EXCE_NULL_GC_REF, true,
+                            cmp_gc_obj, check_gc_obj_succ))
+        goto fail;
+
+    return true;
+fail:
+    return false;
+}
+
+static bool
+aot_call_wasm_struct_obj_new(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
+                             LLVMValueRef rtt_type, LLVMValueRef *struct_obj)
+{
+    LLVMValueRef param_values[2], func, value, res;
+    LLVMTypeRef param_types[2], ret_type, func_type, func_ptr_type;
+
+    param_types[0] = INT8_PTR_TYPE;
+    param_types[1] = INT8_PTR_TYPE;
+    ret_type = GC_REF_TYPE;
+
+    GET_AOT_FUNCTION(wasm_struct_obj_new, 2);
+
+    /* Call function wasm_struct_obj_new() */
+    param_values[0] = func_ctx->exec_env;
+    param_values[1] = rtt_type;
+    if (!(res = LLVMBuildCall2(comp_ctx->builder, func_type, func, param_values,
+                               2, "call"))) {
+        aot_set_last_error("llvm build call failed.");
+        goto fail;
+    }
+
+    *struct_obj = res;
+    return true;
+fail:
+    return false;
+}
+
+static bool
+aot_call_wasm_struct_obj_set_field(AOTCompContext *comp_ctx,
+                                   AOTFuncContext *func_ctx,
+                                   LLVMValueRef struct_obj,
+                                   LLVMValueRef field_idx,
+                                   LLVMValueRef field_value)
+{
+    LLVMValueRef param_values[3], func, value, field_value_ptr;
+    LLVMTypeRef param_types[3], ret_type, func_type, func_ptr_type;
+
+    if (!(field_value_ptr = LLVMBuildAlloca(
+              comp_ctx->builder, LLVMTypeOf(field_value), "field_value_ptr"))) {
+        aot_set_last_error("llvm build alloca failed.");
+        goto fail;
+    }
+    if (!LLVMBuildStore(comp_ctx->builder, field_value, field_value_ptr)) {
+        aot_set_last_error("llvm build store failed.");
+        goto fail;
+    }
+    if (!(field_value_ptr =
+              LLVMBuildBitCast(comp_ctx->builder, field_value_ptr,
+                               INT8_PTR_TYPE, "field_value_ptr"))) {
+        aot_set_last_error("llvm build bitcast failed.");
+        goto fail;
+    }
+
+    param_types[0] = INT8_PTR_TYPE;
+    param_types[1] = I32_TYPE;
+    param_types[2] = INT8_PTR_TYPE;
+    ret_type = VOID_TYPE;
+
+    GET_AOT_FUNCTION(wasm_struct_obj_set_field, 3);
+
+    /* Call function wasm_struct_obj_set_field() */
+    param_values[0] = struct_obj;
+    param_values[1] = field_idx;
+    param_values[2] = field_value_ptr;
+    if (!LLVMBuildCall2(comp_ctx->builder, func_type, func, param_values, 3,
+                        "call")) {
+        aot_set_last_error("llvm build call failed.");
+        goto fail;
+    }
```

Comment:
Had better re-implement this function and aot_call_wasm_struct_obj_get_field, don't BuildAlloc and call runtime API to access struct field, just directly load/store struct field based on the field type/value, so as to improve performance.